internal/errors/errors_test.go

```diff
@@ -0,0 +1,1498 @@
+				args: args{
+					err: errors.New("logging retry"),
+					ref: func() reflect.Value {
+						var i int
+						return reflect.ValueOf(&i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to output  logs, retrying...: logging retry")
+			return test{
+				name: "return an ErrLoggingRetry error when err is not empty and ref is empty.",
+				args: args{
+					err: errors.New("logging retry"),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to output  logs, retrying...")
+
+			return test{
+				name: "return an ErrLoggingRetry error when err is empty and ref is not empty.",
+				args: args{
+					ref: func() reflect.Value {
+						var i int
+						return reflect.ValueOf(&i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrLoggingRetry(test.args.err, test.args.ref)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrLoggingFailed(t *testing.T) {
+	type args struct {
+		err error
+		ref reflect.Value
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		func() test {
+			wantErr := errors.New("failed to output  logs: logging retry")
+			return test{
+				name: "return an ErrLoggingFailed error when err and ref are not empty.",
+				args: args{
+					err: errors.New("logging retry"),
+					ref: func() reflect.Value {
+						var i int
+						return reflect.ValueOf(&i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to output  logs: logging retry")
+			return test{
+				name: "return an ErrLoggingFailed error when err is not empty and ref is empty.",
+				args: args{
+					err: errors.New("logging retry"),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
+		func() test {
+			wantErr := errors.New("failed to output  logs")
+			return test{
+				name: "return an ErrLoggingFailed error when err is empty and ref is not empty.",
+				args: args{
+					ref: func() reflect.Value {
+						var i int
+						return reflect.ValueOf(&i)
+					}(),
+				},
+				want: want{
+					wantErr,
+				},
+			}
+		}(),
```

Comment:
can you add a test case that both ref and err are empty?

Reply:
I already added